### ISSUE TITLE
"cargo doc": error: could not document `rustix` v0.36.11

### ISSUE THREAD

**Issue by fekerr** [rust-lang/rust#109631]
Experimenting with Rocket and learning Rust. Updating this issue is a work in progress.

fekerr@LAPTOP:~/src/boing001$ cat Cargo.toml
``` text 
[package]
name = "boing001"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
# rocket = "0.4.11"
rocket = "0.5.0-rc.3"

```

### Code

```Rust
<code>
fekerr@LAPTOP:~/src/boing001$ cat src/main.rs
// 20230325 fekerr
// main.rs
//
// traitobj crate - issues in future?

// Rocket requires rustc nightly, and these two lines.
#![feature(proc_macro_hygiene, decl_macro)]
#[macro_use] extern crate rocket;

fn main() {
    println!("Hello, world!");
}
```

### Meta

`rustc --version --verbose`:
```
 rustc --version --verbose
rustc 1.70.0-nightly (1459b3128 2023-03-23)
binary: rustc
commit-hash: 1459b3128e288a85fcc4dd1fee7ada2cdcf28794
commit-date: 2023-03-23
host: x86_64-unknown-linux-gnu
release: 1.70.0-nightly
LLVM version: 15.0.7

```

### Error output

```
<output>

cargo doc 2> cargo_doc.txt
[cargo_doc.txt](https://github.com/rust-lang/rust/files/11071785/cargo_doc.txt)


```

<details><summary><strong>Backtrace</strong></summary>
<p>

```

<backtrace>

fekerr@LAPTOP:~/src/boing001$ RUST_BACKTRACE=1 cargo build 2> cargo_build_rbt1.txt
fekerr@LAPTOP:~/src/boing001$ RUST_BACKTRACE=1 cargo build 2> cargo_doc_rbt1.txt
[cargo_build_rbt1.txt](https://github.com/rust-lang/rust/files/11071820/cargo_build_rbt1.txt)
[cargo_doc_rbt1.txt](https://github.com/rust-lang/rust/files/11071821/cargo_doc_rbt1.txt)

```

</p>
</details>

**Comment by fekerr** [comment]
Hello! I'll preserve my folder and proceed on to "boing002". :) 

Thanks!

**Comment by fekerr** [comment]
private github here (can make public or invite collab, whichever is "easier")
https://github.com/fekerr/boing001

**Comment by clubby789** [comment]
Looks like a duplicate of #109614/#109424

**Comment by GuillaumeGomez** [comment]
https://github.com/rust-lang/rust/pull/109330 should fix this bug. Just waiting for a final review.

**Comment by petrochenkov** [comment]
Not fixed by https://github.com/rust-lang/rust/pull/109330, but fixed by https://github.com/rust-lang/rust/pull/109500.

**Comment by GuillaumeGomez** [comment]
It'll be closed once https://github.com/rust-lang/rust/pull/109500 is merged. But I'm surprised it got fixed. When I tested it with my changes, it still failed.

**Comment by clubby789** [comment]
`cargo update` may have caused an update to a dependency so you no longer had the buggy rustix

**Comment by fekerr** [comment]
diff Cargo.toml ../boing002/Cargo.toml
9d8
< # rocket = "0.4.11"
10a10,12
> serde = { version = "1.0", features = ["derive"] }
> serde_json = "1.0"
> fnv = "1.0"

**Comment by fekerr** [comment]
cargo doc fails in this tree:

:~/src/boing001$ cat Cargo.toml
[package]
name = "boing001"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
# rocket = "0.4.11"
rocket = "0.5.0-rc.3"

**Comment by fekerr** [comment]
Does not fail in this tree:

~/src/boing002$ cat Cargo.toml
[package]
name = "boing001"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
rocket = "0.5.0-rc.3"
serde = { version = "1.0", features = ["derive"] }
serde_json = "1.0"
fnv = "1.0"

**Comment by fekerr** [comment]
main.rs in boing001 is just this:

// 20230325 fekerr
// main.rs
//
// traitobj crate - issues in future?

// Rocket requires rustc nightly, and these two lines.
#![feature(proc_macro_hygiene, decl_macro)]
#[macro_use] extern crate rocket;

fn main() {
    println!("Hello, world!");
}

**Comment by fekerr** [comment]
Maybe I should delete my comments. I tried a few things with previous trees and now it seems I can't get it to fail in a new tree.

**Comment by fekerr** [comment]
Probably if I do a cargo clean in boing001 and then cargo doc it will work.

**Comment by fekerr** [comment]
$ grep -i rustix fekerr\@LAPTOP-5JODI0MG\ ~srcboing003src.txt
    Updating rustix v0.37.5 -> v0.37.6
  Downloaded rustix v0.37.6
   Compiling rustix v0.37.6
 Documenting rustix v0.37.6
   Compiling rustix v0.37.6
   Compiling rustix v0.36.11
 Documenting rustix v0.36.11
error: could not document `rustix`
  process didn't exit successfully: `rustdoc --edition=2018 --crate-type lib --crate-name rustix /home/fekerr/.cargo/registry/src/index.crates.io-6f17d22bba15001f/rustix-0.36.11/src/lib.rs --cap-lints allow -o /home/fekerr/src/boing001/target/doc --cfg 'feature="default"' --cfg 'feature="fs"' --cfg 'feature="io-lifetimes"' --cfg 'feature="libc"' --cfg 'feature="std"' --cfg 'feature="termios"' --cfg 'feature="use-libc-auxv"' --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --diagnostic-width=108 -C metadata=ff3e1caa83851d6d -L dependency=/home/fekerr/src/boing001/target/debug/deps --extern bitflags=/home/fekerr/src/boing001/target/debug/deps/libbitflags-f75283d3dcd3daad.rmeta --extern io_lifetimes=/home/fekerr/src/boing001/target/debug/deps/libio_lifetimes-0e896a4daa128e32.rmeta --extern libc=/home/fekerr/src/boing001/target/debug/deps/liblibc-8aab922602f2fc15.rmeta --extern linux_raw_sys=/home/fekerr/src/boing001/target/debug/deps/liblinux_raw_sys-a31463b73c270e13.rmeta --crate-version 0.36.11 --cfg rustc_attrs --cfg linux_raw --cfg core_intrinsics --cfg asm` (exit status: 101)
   Compiling rustix v0.37.6
 Documenting rustix v0.37.6
   Compiling rustix v0.37.6
 Documenting rustix v0.37.6
   Compiling rustix v0.37.6
 Documenting rustix v0.37.6

**Comment by fekerr** [comment]
I doubt I'm really contributing anything here, but a cargo new with just rustix dependency. (I need to look up how to find the available versions of rustix if it's worth narrowing down any further.)

```
[package]
name = "rustix_doc_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
# 37.9 works with "cargo doc"
# rustix = "0.37.9"

rustix = "0.36.11"
```
